File: common/security/src/main/java/org/yechan/AccessTokenBlacklist.java
package org.yechan;

import java.time.Duration;

public interface AccessTokenBlacklist {
    void blacklist(String token, Duration ttl);

    boolean contains(String token);
}

final class NoOpAccessTokenBlacklist implements AccessTokenBlacklist {
    static final NoOpAccessTokenBlacklist INSTANCE = new NoOpAccessTokenBlacklist();

    private NoOpAccessTokenBlacklist() {
    }

    @Override
    public void blacklist(String token, Duration ttl) {
    }

    @Override
    public boolean contains(String token) {
        return false;
    }
}

File: common/security/src/main/kotlin/org/yechan/AccessTokenBlacklist.kt
package org.yechan

import java.time.Duration

interface AccessTokenBlacklist {
    fun blacklist(
        token: String,
        ttl: Duration,
    )

    fun contains(token: String): Boolean
}

object NoOpAccessTokenBlacklist : AccessTokenBlacklist {
    override fun blacklist(
        token: String,
        ttl: Duration,
    ) {
    }

    override fun contains(token: String): Boolean = false
}
